flocker/node/_docker.py

```diff
@@ -442,10 +442,18 @@ def _add():
             # stop on this container Docker might well complain it knows
             # not the container of which we speak. To prevent this we poll
             # until it does exist.
-            while not self._blocking_exists(container_name):
-                sleep(0.001)
-                continue
+            self._blocking_wait(5, self._blocking_exists, container_name)
+
             self._client.start(container_name)
+            # Tests fail intermittently because containers cannot be
+            # removed after being stopped.  It may be that the stop
+            # command sees the container before it is fully started,
+            # and responds with OK, since the container is not running.
+            # This would match with the experience above, where create
+            # returns before the container is fully created.
+            # This is currently only a theory, as the error only occurs
+            # on Buildbot, and cannot be reliably re-created.
```

Comment:
Can we instrument the code/tests so that a failing run on Buildbot reveals the exact request/response information exchanged with Docker?